src/Compilers/CSharp/Portable/BoundTree/BoundDagTest.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using Roslyn.Utilities;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    partial class BoundDagTest
+    {
+        public override bool Equals([NotNullWhen(true)] object? obj) => obj is BoundDagTest other && this.Equals(other);
```

Comment:
it's expected that we would learn that 'obj' is not null when an override of `object.Equals(object)` returns true, even without the attribute. Is that not the case now?

Reply:
I have no idea.  This was essentially copied from elsewhere in the compiler.  If we eliminate this attribute for this use, I would expect we'd do that here and elsewhere.

---
In reply to: [385934220](https://github.com/dotnet/roslyn/pull/42012#discussion_r385934220) [](ancestors = 385934220)